### PR TITLE
Location points

### PR DESCRIPTION
**Pull request by Patrick-Duvall**
This Pr
-Adds leaflet to main page, currently in brokenland as we refactor api 
-Writes api landmark spec, add landmark model, add landmark factory
-Adds landmark serializer
-Adds landmark api routes
-Adds landmarks api controller

### CODE REVIEW

**Comment by Patrick-Duvall** [comment]
Adds Js to populate map when user allows location services

File: app/controllers/landmarks_controller.rb [hunk 1]
```diff
@@ -33,9 +33,20 @@ def index
       nearby_results = JSON.parse(nearby_response.body, symbolize_names: true)[:results]
 
       # Return an array of landmark objects
+      require "pry"; binding.pry
```
Comment (MillsProvosty) [inline app/controllers/landmarks_controller.rb]: take out pry

File: app/controllers/landmarks_controller.rb [hunk 2]
```diff
@@ -33,9 +33,20 @@ def index
...
       @landmarks = nearby_results.map do |landmark_data|
         Landmark.new(landmark_data)
       end
+      @parks = @landmarks.select{|landmark| landmark.types == ["park recreation area"]}
+      @museums = @landmarks.select{|landmark| landmark.types == ["museum"]}
+      @theaters = @landmarks.select{|landmark| landmark.types == ["theater"]}
```
Comment (MillsProvosty) [inline app/controllers/landmarks_controller.rb]: We call it "Theatre", but ok.

File: app/views/landmarks/index.html.erb [hunk 3]
```diff
@@ -1,13 +1,143 @@
...
+  if (category === 'park') {
+    return greenIcon
+  } else if (category ==='theater') {
+    return violetIcon
+  } else {
+    return orangeIcon
+  }
+};
+//what to do after user enables location service
+function onLocationFound(e) {
```
Comment (MillsProvosty) [inline app/views/landmarks/index.html.erb]: What is "e"?

File: app/controllers/landmarks_controller.rb (hunk 2 above)
Comment (Patrick-Duvall) [reply app/controllers/landmarks_controller.rb]: Unless theatre is used in the proper name of a building, production company, etc., theater is the preferred spelling in American English

File: spec/features/landmarks/index_spec.rb [hunk 4]
```diff
@@ -2,17 +2,17 @@
 
 feature 'landmark index' do
   describe 'as a visitor' do
-    scenario 'I can search landmarks with name of city' do
+    xit 'I can search landmarks with name of city' do
```
Comment (MillsProvosty) [inline spec/features/landmarks/index_spec.rb]: Is this supposed to be skipped?

Comment (MillsProvosty) [reply spec/features/landmarks/index_spec.rb]: If they're not being used, consider deleting

**Pull request merged**